.pre-commit-config.yaml

```diff
@@ -50,3 +50,11 @@ repos:
     rev: "3.9.2"
     hooks:
       - id: flake8
+
+  # Create fresh requirements.txt with pip-compile
+  - repo: https://github.com/jazzband/pip-tools
+    rev: 6.4.0
+    hooks:
+      - id: pip-compile
+        files: \setup.(py)$
+        types: ["file", "non-executable", "python", "text"]
```

Comment:
Before each commit, one should create a new requirements.txt, from the requirements in setup.py?

I don't understand this.

Reply:
I think there is relevance for a frozen requirements.txt file with specific versions to represent the state of a built docker image for example, but this is a library that is to be installed in many different environments. Due to that, I think the point of having a very tightly described python environment won't be relevant.